code/game/machinery/vending/vending.dm

```diff
@@ -581,7 +592,22 @@
 
 			if(istype(item_to_stock, /obj/item/ammo_magazine))
 				var/obj/item/ammo_magazine/A = item_to_stock
-				if(A.current_rounds < A.max_rounds)
+				//to_chat(user, "<span class='warning'>DEBUG: Magazine Name: [A]. Recharge?: [recharge]. Current Charge: [machine_current_charge].</span>")
+				if(istype(A, /obj/item/ammo_magazine/lasgun) && recharge)
```

Comment:
If you are using `istype()` chances are you should be doing a more specific proc instead.

Reply:
No, I meant something like /obj/machinery/vending/your_vending_machine/MouseDrop_T